ceph-osd/lib/charms_ceph/utils.py

```diff
@@ -2893,10 +2893,22 @@ def get_osd_state(osd_num, osd_goal_state=None):
     :param osd_num: the OSD id to get state for
     :param osd_goal_state: (Optional) string indicating state to wait for
                            Defaults to None
+    :param timeout: Maximum time in seconds to wait (default: 120)
```

Comment:
Given that this problem was never really seen in any working deployments (except the reported LP bug where OSDs were somehow wedged), maybe we can be a bit more conservative without timeout like 10mins?